### ISSUE TITLE
Implement File modified time

### ISSUE THREAD

**Issue by martinsmmm** [Azure/azure-storage-fuse#95]
Thanks for making the fuse driver.

Any chance to provide some update about 'if' and 'when' the attributes will be available ? What is the plan ?

We are particularly interested in file modification time.

> Some file system APIs have not been implemented: readlink, symlink, link, chmod, chown, fsync, lock and extended attribute calls.

Thank you!

**Comment by cvalla** [comment]
Hello and thank you for your great job.
Files date attribute are all set to 1 gen 1970 when you mount the azure storage.
On Ubuntu 16.04 Installed blobfuse from the microsoft deb repository yesterday.
mounted on /Azure

ls -latr command output is like:

-rwxrwx---  1 root root  858680 gen  1  1970 FILE1.zip
-rwxrwx---  1 root root 1001214 gen  1  1970 FILE2.zip

**Comment by seguler** [comment]
Oh you are right. I was confused. We implemented timestamp support in this PR #29 but never merged... We should implement this.

**Comment by seguler** [comment]
I have implemented this in PR #102 - I am not sure if this PR will make to a release, but test and let us know here.

One caveat is that files timestamps are retrieved from Blob Last-Modified time, however for directories there is no last-modified time on Blob storage. So I am showing the time of list operation for directories.

**Comment by martinsmmm** [comment]
Hi @seguler / all,
Sorry for slow response - get sick last week. Great news!

I will test it during tomorrow and share results. As per your description this is what we are looking for.

Our use case is: We are using Blobstorage as backend/storage for FTP solution. In some cases we need attributes to know which files (luckily not directories) are new (since 'last time') for processing pipeline (Hello Biztalk).

Thank you,
Martin

**Comment by martinsmmm** [comment]
Hi all,
Tested the PR #102 .

The attributes are showing correctly. Thanks @seguler  

```
root@xxx:/xxx/xxx/xxx/xxxtest# ls -la
total 0
drwxrwx--- 3 root root       4096 Feb 20 03:24 .
drwxrwx--- 3 root root       4096 Feb 20 03:24 ..
-rwxrwx--- 1 root root   10485760 Feb  5 03:45 10MB.zip
drwxrwx--- 3 root root       4096 Feb 20 03:24 380855
-rwxrwx--- 1 root root 3006922752 Feb  9 20:02 box-disk1.vmdk
-rwxrwx--- 1 root root          0 Feb  5 03:45 empty-file.txt
drwxrwx--- 3 root root       4096 Feb 20 03:24 test
-rwxrwx--- 1 root root          0 Feb  5 03:46 testfile

```
1970 is gone (https://github.com/Azure/azure-storage-fuse/issues/95#issuecomment-365201365).

We will await merge / release.

Huge Thanks!
Martin

**Comment by martinsmmm** [comment]
Just additional note for folders modification time.

Would it be possible to re-use the information which is displayed on Azure portal UI as '.directory' ?
See the modified column on screenshots.

Maybe, this is possibly overlooked low hanging fruit. (?)

Thanks,
Martin

![screenshot from 2018-02-22 00-12-11](https://user-images.githubusercontent.com/36430903/36521559-422c496e-1765-11e8-961d-16e765d6929f.png)
![screenshot from 2018-02-22 00-12-02](https://user-images.githubusercontent.com/36430903/36521562-4b418ed8-1765-11e8-8387-265a8955d268.png)
![screenshot from 2018-02-22 00-11-50](https://user-images.githubusercontent.com/36430903/36521568-540c7f46-1765-11e8-8572-8f4381a3fe8b.png)

**Comment by seguler** [comment]
This makes sense! I'll look into the feasibility in the code

**Comment by seguler** [comment]
So - I wanna leave the directory timestamps to later... Currently we do not check the existence of .directory file and instead do list_blobs with the prefix to determine if a directory exists or not. Because of this, the code is not very suitable to report the directory timestamps at the moment. 

Will keep this issue open until we figure out a way to report directory timestamps properly.

**Comment by seguler** [comment]
I'll close this as File modified time is implemented in blobfuse 0.3. Will open another one for directory timestamps.

**Comment by hafizullah** [comment]
@seguler is there a PR open for the directory timestamp?
